src/coreclr/jit/lsra.h

```diff
@@ -2064,15 +2066,43 @@ class LinearScan : public LinearScanInterface
     //
     static regMaskTP calleeSaveRegs(RegisterType rt)
     {
-        return varTypeIsIntegralOrI(rt) ? RBM_INT_CALLEE_SAVED : RBM_FLT_CALLEE_SAVED;
```

Comment:
This, and `callerSaveRegs` below, was interestingly doing a `IsType` rather than a `UsesReg` check. This meant that before this PR, `TYP_STRUCT` would hit thee `RBM_FLT_CALLEE_SAVED` path, which isn't really correct.

It doesn't seem to actually impact codegen at all, from what I could tell. But, it's something worth calling out.

Reply:
Also notably, this is the one change that also impacts other platforms (that is it also impacts Arm/Arm64).

Reply:
interestingly - Replacing `((varTypeClassification[TypeGet(vt)] & (VTF_INT | VTF_I)) != 0);` with `varTypeRegister[TypeGet(vt)] == VTR_INT;` is causing TP diff of up to `0.11%`?

Reply:
Yes. You can see that in the isolated PR here: https://dev.azure.com/dnceng-public/public/_build/results?buildId=348931&view=ms.vss-build-web.run-extensions-tab (from https://github.com/dotnet/runtime/pull/89349).